### ISSUE TITLE
New industry shapes need to be traced

### ISSUE THREAD

**Issue by scottanderson** [scottanderson/railroad.studio#103]
The new industry types, such as `CattleFarm`, `MeatPackingPlant`, `Sandhouse`, and `watertower_drgw` have a [placeholder SVG shapes](https://github.com/scottanderson/railroad.studio/blob/1a1339ba51037e6d043f26af610258baa6f20a00/ts/IndustryName.ts#L54) associated with them.

**Comment by illudedPerception** [comment]
i guess water tower drgw can keep the place holder svg looks fine to me on the map (its not that much bigger then the original round one)
so you need cattle farm, wheat farm, packing plant and water well, cause sand house is sand house

**Comment by illudedPerception** [comment]
![image](https://github.com/scottanderson/railroad.studio/assets/126168696/348b2b07-0116-4862-b88a-df733c6bdac3)
this is the sand house, there is no other

**Comment by illudedPerception** [comment]
new industries svg's (loading and unloading have been devided to two different colors black and blue at this moment + lake valley contour map svg (with lake) and test save with new industries placed the same way the images were taken and converted to svg's + Eg image for refrence
[new industries + lake valley contour map svg.zip](https://github.com/scottanderson/railroad.studio/files/13849495/new.industries.%2B.lake.valley.contour.map.svg.zip)

**Comment by scottanderson** [comment]
Although the sand house model did not change, it was converted from a bespoke object type to an industry. RS did not previously render the bespoke object, and is using a placeholder for the new industry version of the sand house.

**Comment by illudedPerception** [comment]
ahaa well its a bit late to do it today cause i need to be up early and i will be back tuesday evening and i will trace it then

**Comment by scottanderson** [comment]
If the SVG paths are already in centimeter scale, you can submit a pull request to modify [industries.ts](https://github.com/scottanderson/railroad.studio/blob/c5683519c2094d8533ff44f327fd6f783c9bc05e/ts/industries.ts#L246) to include the relevant shapes. In this example, `pond` is set as the css class name for the shape, and the style is set in [style.css](https://github.com/scottanderson/railroad.studio/blob/c5683519c2094d8533ff44f327fd6f783c9bc05e/railroad.studio/style.css#L438).

```ts
        'pond': [
            ['M', 2800, 980],
            ['C', 2010, 640, 1600, 500, 1180, 420],
            ['S', 580, 500, -270, 1100],
            ['S', -800, 1420, -1280, 2270],
            ['S', -1330, 3170, -980, 4170],
            ['S', -320, 4910, 570, 5220],
            ['S', 1250, 5940, 2100, 5840],
            ['S', 2640, 5310, 2750, 4360],
            ['Z'],
        ] as PathCommand[],
```

If you need to measure an industry, check out the industry measurement layer. Click on the industry arrow and drag to view coordinates in the industry's coordinate space. Coordinates are logged to the F12 console, so you can trace out several coordinates and easily copy the coordinates from the console output.

**Comment by illudedPerception** [comment]
well i just checked and some of them are in inch scale for some reason and other are mm what is my bad cause i remembered that the game used mm but actually uses cm, any ways its gonna take time for me to re do the map file

**Comment by illudedPerception** [comment]
well i guess i need to select and place them while facing 0° (west)(fixed), well changed the scale in gimp the svg comes out huge, but as i transfer the cords to the industries.ts its still displays ultra small
![image](https://github.com/scottanderson/railroad.studio/assets/126168696/b8635683-70b5-4bcc-9bff-fca71b326cad)

**Comment by scottanderson** [comment]
It looks like the scale might be meter, not centimeter. You could move the decimal places over a couple of spaces (multiply by 100x) for every coordinate and see if that fixes it. You can also use the industry measurement tool from the layers menu to get a feel for what the numbers should be.

![image](https://github.com/scottanderson/railroad.studio/assets/662325/ca06628c-4451-4747-8492-73078351e597)

**Comment by illudedPerception** [comment]
i had bit of time to mess with it so, il prob give a try to the built in function to trace them (as i get time), tho tracing them in gimp is easy but making them work is a headache, also i had to use the console to edit the allmost right scale trace to be where the industry actually is

about the gimp problem
i found the problem its that the svg i save uses a line to define the scale maybe a GIMP problem (idk), don't know what other program to use, tho i can play with it making the image bigger and playing with the px/cm also
other problems i have is that even if i trace them from an image they might be off from some sides depending from what place i took the image in

**Comment by scottanderson** [comment]
Thanks for trying - I would love to improve the development experience around this to make it easier to add nicer and more complex shapes; it simply has not been a priority before now. The current shapes were traced using that tool, but this results in relatively basic forms. It would be better if there were a way to import from the SVG files that you created, and adjust their scale/offset/rotation as needed.

**Comment by illudedPerception** [comment]
well scale and offset adjustment is enough i just messed up the rotation by placing them when facing north

is it possible to use an svg file for the map (scaling wise) ?  i want to make it again because the map in the zip file is mm scale but i want to see if i can get it to cm scale

**Comment by scottanderson** [comment]
Loading SVG asset files is technically possible, but I don't know if you can specify CSS style names from within Gimp, and I am unsure if it's even possible to reference CSS styles from an SVG asset at all. Ideally, I want to avoid using inline styles because dark mode is implemented as a CSS selector.

If you can figure out some way to reference CSS styles, such as `pond` from my example above, within Gimp, then I would be willing to experiment with adding support for SVG assets. Otherwise, I think the paths will need to be converted to RS's arcane format.

**Comment by illudedPerception** [comment]
simplelified it looks like this

      width="xx.xxcm" height="xx.xxcm"
      viewBox="0 0 0 0"
 <path id="building" So this should me possible to edit with the styles ?
        fill="blue" stroke="gold" stroke-width="1"
        "path code" />

<path id="platfrom"
        fill="black" stroke="gold" stroke-width="1"
        "path code" />

**Comment by scottanderson** [comment]
Here is an example of how the Freight Depot appears in the DOM. Note how the paths only have `d` and `class` attributes. All styling information is handled in CSS.

```html
<g transform="translate(4660 9236) rotate(90.00)" class="industry freightdepot">
  <title>Freight Depot</title>
  <path d="M2200 -1000H-3300V-700H-2300V1000H2200Z " class="platform"></path>
  <path d="M2100 -900H-2200V900H2100Z " class="building"></path>
</g>
```

**Comment by scottanderson** [comment]
@illudedPerception, I added experimental support for svg assets in https://github.com/scottanderson/railroad.studio/commit/1592fb77889c8e9adae833d9febf2e6a6634060a with some of the files that you attached. Are you are able to adjust the origin point in your editor to make the industries appear in the correct location relative to the arrows? In the image below, the green box is roughly the location that the image should be. The bounds for the wheat farm, for example, should be roughly (-1800,-3000) to (3200,3000).

![image](https://github.com/scottanderson/railroad.studio/assets/662325/94f04bc3-117e-46c9-a601-ef8f65e451a2)

**Comment by illudedPerception** [comment]
Should be able to just need to keep im mind that the resolution would not get too big and that the scale is right

**Comment by illudedPerception** [comment]
well the origin i cant change, its the top left corner of the svg viewbox  (il research this tho)

**Comment by scottanderson** [comment]
Thanks for attempting to get it to work anyway. I was able to align the industry with the green rectangle by translating the image in map rendering code, see 135c404. Unfortunately, this requires setting the x/y offset in typescript code. I set the industry root SVG to use `width="5000" height="6000"` to get the wheat farm image to be roughly the correct world size (50m x 60m).

**Comment by illudedPerception** [comment]
Well those industry svg are all in the wrong scale cause of some problem with GIMP i noticed it yesterday also wrong oretation cause of my mistake

**Comment by Tedz62** [comment]
Have you tried Inkscape for working on svg file? Seems to be able to work with css too.

**Comment by scottanderson** [comment]
The issue with using CSS is a technical limitations of the web browser.
Embedded SVGs can not reference CSS rules from the document. Any style
rules that an SVG file uses needs to be defined in the same file.

**Comment by illudedPerception** [comment]
> Have you tried Inkscape for working on svg file? Seems to be able to work with css too.

i looked into it before this comment, but non of that program made sense and there was other problems with it, tho i did get a test file out still not any better then what i used to make the test SVG's. So better would be to stick to tracing the old way so CSS can be applied to it, also with SVGs the origin point is upper left corner, but the industries rarely have the origin point there (coaling tower has i think)

**Comment by Tedz62** [comment]
I wonder if cad would be good for controlling the origin point. I use QCAD and it has a quick and advanced SVG export. I would think the 0,0 coordinate would be the origin of an SVG. The image below is Advanced SVG export dialog. It's easy to import an image to trace, scale and move the shapes. QCAD has a demo or I can try something if you would like.
![image](https://github.com/scottanderson/railroad.studio/assets/125933603/89ccd75a-7d3a-40bc-81df-2d67c0b04cbc)

**Comment by illudedPerception** [comment]
> I wonder if cad would be good for controlling the origin point. I use QCAD and it has a quick and advanced SVG export. I would think the 0,0 coordinate would be the origin of an SVG. The image below is Advanced SVG export dialog. It's easy to import an image to trace, scale and move the shapes. QCAD has a demo or I can try something if you would like. ![image](https://private-user-images.githubusercontent.com/125933603/306750495-89ccd75a-7d3a-40bc-81df-2d67c0b04cbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1NDYwODEsIm5iZiI6MTcwODU0NTc4MSwicGF0aCI6Ii8xMjU5MzM2MDMvMzA2NzUwNDk1LTg5Y2NkNzVhLTdkM2EtNDBiYy04MWRmLTJkNjdjMGIwNGNiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMVQyMDAzMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YWEzNWQ2ZGRmZTM1ZDFhNDk1NjAzYTYyMzNiYWY3ZTZkYjRmNjdlMWJhMTgwMWFmZjE1NjBhMjk5ZDFlODcwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Bpn8NFG7vB3PCaeq2D1-7SCEiOv71Q_o8mUC7udEAWk)

i googeled about it and basically as i understood it can be in the upper left corner, its been a while since i was troubleshooting it

**Comment by scottanderson** [comment]
Just to clarify what the offset issue is, the HTML `image` element can only render positive XY coordinates. Aligning an `image` element at the industry origin without an offset means industries would have their paths cropped or stretched. To correct for this, any embedded `image` needs a `transform` unique to that file to properly align the image in to user-space (ex: [backgrounds](https://github.com/scottanderson/railroad.studio/blob/fbe709d9b1c6af6eabe6fd58d66b07a2300ba2c6/ts/RailroadMap.ts#L749-L754)).

**Comment by illudedPerception** [comment]
Well anyways imma trace them the old way how its been done so far, but im going trace them in the game with RROx and then make an map image out of it and when i load it in RS so i can see if the trace is on spot and use it to help me trace, this will work until RROx still works after that i don't know

**Comment by illudedPerception** [comment]
im tracing the farm stuff then il move on to gold industry stuff
![image](https://github.com/scottanderson/railroad.studio/assets/126168696/f135bb6c-88ad-4f87-80c7-288c3f1dc9e2)

**Comment by illudedPerception** [comment]
well im stuck on the meat packing plat with the polyRect i also tried polyRectRel getting the shape needed seems impossible
img what i want it to be
![image](https://github.com/scottanderson/railroad.studio/assets/126168696/8b022dcd-ebbb-4a49-84fd-1c31cab73fd6)

**Comment by scottanderson** [comment]
I think this is resolved with #140. Thanks @illudedPerception
